libraries/AP_Scheduler/AP_Scheduler.tpp

```diff
@@ -0,0 +1,186 @@
+extern const AP_HAL::HAL& hal;
```

Comment:
> @lucasdemarchi are writing all the new structure APM HAL for linux under Intel statement and i don't know why, he must have his reasons.

I don't really understand this statement. It's pretty standard in open source development. Whoever is paying me for the development (Intel in this case) has the copyright, and that's the line that usually goes on top of the license.  The author can be directly tracked by git (and you can see the author of each line with git blame and the history with git log).